holoviews/ipython/widgets.py

```diff
@@ -257,176 +218,11 @@ def get_plot_size():
-            or OutputMagic.options['fig'] =='svg'):
-            self.image_widget = widgets.HTMLWidget()
-        else:
-            self.image_widget = widgets.ImageWidget()
-
-        if self.cached:
-            self.image_widget.value = list(self.frames.values())[0]
-        else:
-            self.image_widget.value = self._plot_figure(0)
-        self.image_widget.set_css(self.css)
-
-        # Initialize interactive widgets
-        interactive_widget = widgets.interactive(self.update_widgets,
-                                                 **self.pwidgets)
-        interactive_widget.set_css(self.css)
-
-        # Display widgets
-        display(interactive_widget)
-        display(self.image_widget)
-        return '' # Suppresses outputting ViewableElement repr when called through hook
-
-
-    def _get_dim_vals(self, indices, idx):
-        """
-        Get the dimension values along the supplied dimension,
-        computed from the supplied indices into the mock_obj.
-        """
-        indices[idx] = slice(None)
-        vals = [k[idx] if isinstance(k, tuple) else k
-                for k in self.mock_obj[tuple(indices)].keys()]
-        return vals
-
-
-    def update_widgets(self, **kwargs):
-        """
-        Callback method to process the new keys, find the closest matching
-        ViewableElement and update all the widgets.
-        """
-
-        # Do nothing if dimension values are unchanged
-        if all(v == self.dim_val[k] for k, v in kwargs.items()):
-            return
-
-        # Place changed dimensions first
-        changed_fn = lambda x: x[1] == self.dim_val[x[0]]
-        dimvals = sorted(kwargs.items(), key=changed_fn)
-
-        # Find the closest matching key along each dimension and update
-        # the matching widget accordingly.
-        checked = [slice(None) for i in range(self.mock_obj.ndims)]
-        for dim, val in dimvals:
-            if not isnumeric(val): val = str(val)
-            dim_idx = self.mock_obj.get_dimension_index(dim)
-            widget = self.pwidgets[dim]
-            vals = self._get_dim_vals(checked, dim_idx)
-            if val not in vals:
-                if isnumeric(val):
-                    val = vals[np.argmin(np.abs(np.array(vals) - val))]
-                else:
-                    val = str(vals[0])
-            checked[dim_idx] = val
-            self.dim_val[dim] = val
-            widget.value = round(val, 5) if isnumeric(val) else val
-
-        # Update frame
-        checked = tuple(checked)
-        if self.cached:
-            self.image_widget.value = self.frames[checked]
-        else:
-            self.image_widget.value = self._plot_figure(self.keys.index(checked))
-
-
-
-class ScrubberWidget(NdWidget):
+    NdWidget is an abstract base class implementing a method to find
+    the dimensions and keys of any ViewableElement, GridSpace or
+    UniformNdMapping type.  In the process it creates a mock_obj to
+    hold the dimensions and keys.
```

Comment:
Why is this class ``Parameterized`` if you aren't declaring any parameters? Maybe some of you ``self`` attribute should be parameters instead?